src/components/structures/InteractiveAuth.js

```diff
@@ -61,11 +79,19 @@ export default React.createClass({
             authData: this.props.authData,
             doRequest: this._requestCallback,
             startAuthStage: this._startAuthStage,
+            inputs: this.props.inputs,
+            stateUpdated: this._authStateUpdated,
+            makeRegistrationUrl: this.props.makeRegistrationUrl,
+            matrixClient: this.props.matrixClient,
+            sessionId: this.props.sessionId,
+            clientSecret: this.props.clientSecret,
+            emailSid: this.props.emailSid,
         });
 
         this._authLogic.attemptAuth().then((result) => {
             this.props.onFinished(true, result);
         }).catch((error) => {
+            this.props.onFinished(false, error);
```

Comment:
Hmm, yeah. Seems like the issue is that while the non-dialog version wants the InteractiveAuth to go away, since there's no value in the thing sitting there saying 'error'. This isn't sensible for the dialog version though, and you do want the dialog to say it failed, then you can close it & try again.

So seems like the right fix is to make the dialog version catch onFinished(false) and replace the InteractiveAuth component with an error message?